src/table.ts

```diff
@@ -1904,14 +1904,17 @@ class Table extends common.ServiceObject {
         ? optionsOrCallback
         : ({} as InsertRowsOptions);
     const callback =
-      typeof optionsOrCallback === 'function'
-        ? optionsOrCallback
-        : (cb as InsertRowsCallback);
+      typeof optionsOrCallback === 'function' ? optionsOrCallback : cb;
 
-    this._insertAndCreateTable(rows, options).then(
-      resp => callback(null, resp),
-      err => callback(err, null)
-    );
+    const promise = this._insertAndCreateTable(rows, options);
+    if (callback) {
```

Comment:
I don't think this is needed, as there's already a promisified version being returned. See https://github.com/googleapis/nodejs-bigquery/blob/a138347855f74d4e5c889dc42b00992c4a3808a6/src/table.ts#L2388.

I think the unhandled rejection may come from further down.

Reply:
You're right, this PR fixes nothing then. Easy to miss. Not the biggest fan of code like that tbh.

cc @feywind